core/arch/riscv/include/riscv.h

```diff
@@ -155,6 +155,27 @@ static inline __noprof void wfi(void)
 	asm volatile ("wfi");
 }
 
+static inline __noprof void riscv_cpu_pause(void)
+{
+	unsigned long dummy = 0;
+
+	/*
+	 * Use a divide instruction to force wait
+	 * for multiple CPU cycles.
+	 */
+	asm volatile ("div %0, %0, zero" : "=r" (dummy));
```

Comment:
I was somewhat surprised by this line until I read that "RISC-V doesn’t raise an exception on divide by zero". Perhaps worth adding as a comment here?